api/src/main/java/org/openmrs/validator/DrugOrderValidator.java

```diff
@@ -90,7 +90,7 @@ public void validate(Object obj, Errors errors) {
 			if (order.getAction() != Order.Action.DISCONTINUE) {
 				ValidationUtils.rejectIfEmpty(errors, "dosingType", "error.null");
 			}
-			if (order.getDrug() == null || order.getDrug().getConcept() == null) {
+			if (!order.isNonCodedDrug() && (order.getDrug() == null || order.getDrug().getConcept() == null)) {
```

Comment:
This change feels wrong. Even if it is a non-coded drug, the concept should still be set, at least at the point that it gets to the validator. If not, it may indicate that the change is best done elsewhere rather than in the OrderService.